modules/python3/pythonscript.h

```diff
@@ -24,112 +24,127 @@
+    Can be used to parse results from the script. This callback will only be called of the
+    script executed with out problems
+    *
+    * @return true, if script execution has been successful
     */
-    class IVW_MODULE_PYTHON3_API PythonScript {
-
-    public:
-        using VariableMap = std::unordered_map<std::string, pybind11::object>;
-
-        PythonScript();
-
-        /**
-        * Frees the stored byte code. Make sure that the
-        * Python interpreter is still initialized
-        * when deleting the script.
-        */
-        virtual ~PythonScript();
-
-        /**
-        * Sets the source for the Python (replacing the current source).
-        */
-        void setSource(const std::string& source);
-
-        /**
-        * Returns the script's source.
-        */
-        std::string getSource() const;
-
-        /**
-        * Runs the script once,
-        * if the script has changed since last compile a new compile call will be issued.
-        *
-        * If an error occurs, the error message is logged to the inviwo logger and python standard
-        output.
-        *
-        * @param extraLocalVariables a map of  keys and PyOjbects that will available as local
-        variables in the python scripts, eg {"number" , PyValueParser::toPyObject<int>(123) } will
-        be avaible as the local variable 'number' in the script
-        * @param callback a callback that will be called once the script has finished executing. The
-        callback takes a PyObject representing the python dict of local variables from the script.
-        Can be used to parse results from the script. This callback will only be called of the
-        script executed with out problems
-        *
-        * @return true, if script execution has been successful
-        */
-        bool run(const VariableMap& extraLocalVariables = VariableMap(),
-            std::function<void(pybind11::dict)> callback = [](pybind11::dict dict) {});
-
-
-        bool run(std::function<void(pybind11::dict)> callback) {
-            return run({}, callback);
-        }
-
-        void setFilename(std::string filename);
-
-    private:
-        bool checkCompileError();
-        bool checkRuntimeError();
-
-        /**
-        * Compiles the script source to byte code, which speeds up script execution. This function
-        * is called by ::run when needed (eg. the source code has changed)
-        *
-        * @return true, if script compilation has been successful
-        */
-        bool compile();
-
-        std::string source_;
-        std::string filename_;
-        void* byteCode_;
-        bool isCompileNeeded_;
-    };
+    bool run(const VariableMap& extraLocalVariables = VariableMap(),
+             std::function<void(pybind11::dict)> callback = [](pybind11::dict dict) {});
+
+    bool run(std::function<void(pybind11::dict)> callback) { return run({}, callback); }
+
+    virtual void setFilename(const std::string& filename);
```

Comment:
Ideally, file location should not be needed at script level. Not related to this PR I guess

Reply:
that was already there, added by @r-englund earlier.

Reply:
I am a bit torn here. I can see use cases for both a string/script property as well as an external file.